msprime/demography.py

```diff
@@ -2431,6 +2459,301 @@ def _american_admixture_model():
         demography.add_population_split(5920, derived=["AMH"], ancestral="ANC")
         return demography
 
+    def to_demes(self) -> demes.Graph:
+        """
+        Creates a :class:`demes.Graph` object from the demography.
+
+        The returned graph can be saved as a
+        `Demes-format <https://popsim-consortium.github.io/demes-spec-docs/main/>`_
+        YAML file using the :ref:`demes <demes:sec_introduction>` API.
+
+        .. code::
+
+            import demes
+
+            demography = msprime.Demography.island_model([1000] * 3, 1e-5)
+            graph = demography.to_demes()
+            demes.dump(graph, "island_model.yaml")
+
+        Or plotted using the `demesdraw <https://github.com/grahamgower/demesdraw>`_
+        visualisation package.
+
+        .. code::
+
+            import demesdraw
+
+            demography = msprime.Demography.island_model([1000] * 3, 1e-5)
+            graph = demography.to_demes()
+            ax = demesdraw.tubes(graph)
+            ax.figure.savefig("island_model.pdf")
+
+        :return: A :class:`demes.Graph` object corresponding to the demography.
+        :rtype: demes.Graph
+        """
+        resolved = self.validate()
+        b = demes.Builder()
+        for pop in resolved.populations:
+            initial_epoch = dict(
+                end_size=pop.initial_size, growth_rate=pop.growth_rate, end_time=0
+            )
+            b.add_deme(
+                pop.name,
+                description=pop.description if pop.description else None,
+                start_time=math.inf,
+                epochs=[initial_epoch],
+            )
+
+        deme_map = dict()
+        for pop in resolved.populations:
+            deme_map[pop.name] = b.data["demes"][pop.id]
+            deme_map[pop.id] = b.data["demes"][pop.id]
```

Comment:
Why store both? Probably clearer to do something like ``resolved[population].name`` to be certain that we're converting to names.